Explorer/Assets/DCL/Multiplayer/Movement/Systems/RemotePlayerAnimationSystem.cs

```diff
@@ -0,0 +1,156 @@
+            view.SetAnimatorBool(AnimationHashes.LONG_FALL, animState.IsLongFall);
+
+            animationComponent.States = animState;
+        }
+
+        private static void UpdateAnimatorBlends(IAvatarView view, in AnimationStates animStates)
+        {
+            if (!animStates.IsGrounded)
+                return;
+
+            view.SetAnimatorFloat(AnimationHashes.MOVEMENT_BLEND, animStates.MovementBlendValue > BLEND_EPSILON ? animStates.MovementBlendValue : 0f);
+            view.SetAnimatorFloat(AnimationHashes.SLIDE_BLEND, animStates.SlideBlendValue > BLEND_EPSILON ? animStates.SlideBlendValue : 0f);
+        }
+
+        private static void InterpolateAnimations(IAvatarView view, ref CharacterAnimationComponent anim, in InterpolationComponent intComp)
+        {
+            if (!anim.States.IsJumping && intComp.End.animState.IsJumping && Mathf.Abs(intComp.Start.position.y - intComp.End.position.y) > JUMP_EPSILON)
+                AnimateFutureJump(view, ref anim, intComp.End.animState);
+
+            AnimationStates startAnimStates = intComp.Start.animState;
+            AnimationStates endAnimStates = intComp.End.animState;
+
+            bool bothPointBlendsAreZero = startAnimStates.MovementBlendValue < BLEND_EPSILON && endAnimStates.MovementBlendValue < BLEND_EPSILON
+                                                                                             && startAnimStates.SlideBlendValue < BLEND_EPSILON && endAnimStates.SlideBlendValue < BLEND_EPSILON;
+
+            if (bothPointBlendsAreZero && Vector3.SqrMagnitude(intComp.Start.position - intComp.End.position) > MOVEMENT_EPSILON)
+                BlendBetweenTwoZeroMovementPoints(ref anim, intComp);
+            else
+            {
+                anim.States.MovementBlendValue = Mathf.Lerp(startAnimStates.MovementBlendValue, endAnimStates.MovementBlendValue, intComp.Time / intComp.TotalDuration);
+                anim.States.SlideBlendValue = Mathf.Lerp(startAnimStates.SlideBlendValue, endAnimStates.SlideBlendValue, intComp.Time / intComp.TotalDuration);
+            }
+
+            UpdateAnimatorBlends(view, anim.States);
+        }
+
+        private static void AnimateFutureJump(IAvatarView view, ref CharacterAnimationComponent anim, in AnimationStates animState)
+        {
+            anim.States.IsGrounded = animState.IsGrounded;
+            anim.States.IsJumping = animState.IsJumping;
+
+            view.SetAnimatorTrigger(AnimationHashes.JUMP);
+
+            view.SetAnimatorBool(AnimationHashes.GROUNDED, anim.States.IsGrounded);
+            view.SetAnimatorBool(AnimationHashes.JUMPING, anim.States.IsJumping);
+        }
+
+        private static void BlendBetweenTwoZeroMovementPoints(ref CharacterAnimationComponent anim, in InterpolationComponent intComp)
+        {
+            float speed = Vector3.Distance(intComp.Start.position, intComp.End.position) / intComp.TotalDuration;
+
+            float midPointBlendValue = speed > 9.5f ? 3f :
```

Comment:
Can you give meaningful names to these magic numbers?

Reply:
good catch, extracted most of them, but missed that last part 👀